dcs/task.py

```diff
@@ -1077,6 +1082,39 @@ def __eq__(self, other) -> bool:
         return False
 
 
+class CarpetBombing(Task):
+    Id = "CarpetBombing"
+
+    class AttackType(Enum):
+        Carpet = "Carpet",
+
+        def __eq__(self, other: Any) -> bool:
+            if isinstance(other, str):
+                return self.value == other
+            return self == other
```

Comment:
Does this not just infinitely recurse?

```python
>>> class Foo(Enum):
...     A = "a"
...     def __eq__(self, o):
...             if isinstance(o, str):
...                     return self.value == o
...             return self == o
...
>>> Foo.A == 0
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 6, in __eq__
  File "<stdin>", line 6, in __eq__
  File "<stdin>", line 6, in __eq__
  [Previous line repeated 496 more times]
RecursionError: maximum recursion depth exceeded
```

Reply:
> Improving python skills at every step

In case it helps, `a == b` calls `a.__eq__(b)`. If that returns `True` or `False`, that's the end of it. If it returns `NotImplemented` (not to be confused with raising `NotImplementedError`, which is something else), it retries with `b.__eq__(a)` (that last part I actually only learned the other day while looking through the docs for reviewing your other PR, so you'll probably find a bunch of existing implementations here that don't do that).

What you've got here is `a.__eq__(b)` which in turn calls `a.__eq__(b)` unless `b` is an instance of `str` :)

Reply:
@DanAlbert - I simplified the implementation to care only about str and Expend instances, others are not equal.